### DISCUSSION TITLE
Behavior on restart without backend connectivity

### DISCUSSION THREAD

**Discussion by bennydiamond** [Blackymas/NSPanel_HA_Blueprint#2255] [Q&A] [answered]
Hi,

I would like some information on what happens in a particular scenario.

Let's say your Home Assistant and ESPHome instances are down.
Your NSPanel has already been setup with the blueprint beforehand.
You reboot/power cycle your NSPanel.

What's the expected outcome?
Is the current config actually locally saved and brought up on boot from NSPanel?
Will the relay fallback feature work?

I understand that without backend connectivity, there is little to do here. I just want to make sure I'm not locking myself out of controlling the lights physically toggled by the NSPanel.

Thanks

**Comment by edwardtfn** [comment, marked as answer]
Well, quite a lot of things are saved locally, but still quite a lot of things won't work oj that scenario.

First, when the API is unavailable (due to lack of Wi-Fi or Home Assistant being out), the panel will show the last info available until the page is refreshed (by screensaver, or page fallback process, or even user's interaction).
The detailed pages depending on info from Home Assistant will stop workimg (it won't open the pages), including all the button pages, weather pages, etc.
Some passes will still work until the panel reboots, like QR code.
By default, ESPHome will automatically reboot on that situation, in order to tentatively recover the failure. For a wifi outage, we kept the ESPHome default of 15min, but for the API we changed it to 60min, which means your panel will be rebooting in a loop. You can change these intervals or even bypass those as indicated in the documentation for customizations [here](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/customization.md#reboot-when-api-fails).
If after the reboot the panel still cannot access Home Assistant or the automation (Blueprint) isn't working, it will take a few more seconds on the boot screen, waiting for the data, but will then go to the selected wake up page, but with limited information.
The time will still be visible, and with the right format, but unless you have set a [custom time server](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/customization.md#sntp-time-server) and that is available, it will be relying only on ESP32 to maintain that info correct. If the panel isn't out of power for a long interval, I would expect a quite reliable info shown.
The date won't be shown, as that is controlled by the Blueprint (due to translations).
Weather, button pages, entities pages, custom buttons, utilities pages and QR code pages 0 won't be available. And I probably forgot something t9 this list. 😉
Sensor which depends on HA obviously won't be shown.
The basic engine of fallback to home page and then sleep will work normally, with the intervals selected by user.
For internal temperature, the system will fallback to its embedded sensor, with the selected corrections, so that should be visible.
If the hardware buttons are assigned to the physical relay or a fallback is set, that will still work.
The climate add-on will still work and the embedded sensor will be used to control the relays and keep the desired temperatures. You should also be able to open the climate page to change its settings (there's a bug in the with v4.3.11, but a fix is already on v4.3.12.dev to be released soon).
The cover add-on cannot be controlled without the api, yet. I wanna change this in the near future.

Have I answered your question?

**Comment by bennydiamond** [comment]
Hi,

thanks for getting back to me so quickly.
I'm sorry I couldn't return the courtesy...

But yes you answered my question and way more!
I just wanted to make sure I wouldn't lose physical control over my lights if the back-end isn't available.
I'm on the mantra that home automation should complement basic home usage, not create hard dependencies.

Thank you very much. Question answer.
